apps/arweave/src/ar_data_sync_worker_master.erl

```diff
@@ -49,28 +59,31 @@ is_syncing_enabled() ->
 %% @doc Returns true if we can accept new tasks. Will always return false if syncing is
 %% disabled (i.e. sync_jobs = 0).
 ready_for_work() ->
-	{ok, Config} = application:get_env(arweave, config),
-	TotalTaskCount = get_task_count(scheduled) + get_task_count(queued),
-	TotalTaskCount < (Config#config.sync_jobs * 50).
+	try
+		gen_server:call(?MODULE, ready_for_work, 1000)
```

Comment:
Previously when this was implement via an ETS query it could lead to race conditions where the `ar_data_sync_worker_master` was in the middle of processing a message and therefore unable to update the ETS data - during that tim the `ar_data_sync` processes would send thousands of extra tasks. Now that the `read_for_work` is itself implemented as a message it synchronizes the request and forces the `ar_data_sync` processes to wait for the `ar_data_sync_worker_master` to finish processing one batch of tasks before sending more.